Updates for Terraform policy docs (#2864)
hashicorp/web-unified-docs: content/terraform-policy/v0.1.x (beta)/docs/policy/reference/policy/input.mdx

@include "beta.mdx"

## Configuration model

The `input` block supports the following configuration:

- `input` &nbsp "[`<label>`](#label)" &nbsp block | repeatable
  - [`default`](#default) &nbsp any | optional
  - [`description`](#description) &nbsp string | optional
  - [`nullable`](#nullable) &nbsp boolean | optional
  - [`sensitive`](#sensitive) &nbsp boolean | optional
  - [`type`](#type) &nbsp type constraint | required

## Complete configuration example

The following example demonstrates an `input` block with all configuration options specified, a second `input` block, and a `policy` block that references inputs:

<CodeBlockConfig filename="policies/example.policy.hcl">

```hcl
input "allowed_instance_types" {
  type        = list(string)
  description = "List of allowed EC2 instance types"
  default     = ["t3.micro", "t3.small"]
  sensitive   = false
  nullable    = false
}

input "environment" {
  type        = string
  description = "The environment name (For example: dev, staging, prod)"
  default     = "dev"
}

resource_policy "aws_instance" "instance_type_check" {
  enforce {
    condition     = core::contains(input.allowed_instance_types, attrs.instance_type)

>     error_message = "Instance type ${attrs.instance_type} is not in the allowed list for the "${input.environment}" environment: ${core::join(", ", input.allowed_instance_types)}"

Reviewer: Broken HCL: the inner `"${input.environment}"` nests unescaped quotes inside the string, so this example fails to parse.

```suggestion
    error_message = "Instance type ${attrs.instance_type} is not in the allowed list for the ${input.environment} environment: ${core::join(", ", input.allowed_instance_types)}"
```

  }
}
```

</CodeBlockConfig>

## Specification

An `input` block supports the following configuration.

### Label

The input block label gives a local name to the input variable. Must be unique within the policy file. Reference input variables with the syntax `input.<label>`.

### `default`

The `default` attribute declares provides a default value for the input variable. If omitted and no value is provided by the policy configuration, the value is null, and Terraform policy errors unless `nullable` is set to `true`.

- Data type: any
- Default: None

> - Example: [String input with default](#string-input-with-default) 

Reviewer: Typo: "declares provides a default value" -- drop one verb. Also, the example link `#string-input-with-default` is broken now that the heading is "String input with default value" (anchor `#string-input-with-default-value`).

## Configuration model

The `input` block supports the following configuration:

- `input` &nbsp "[`<label>`](#label)" &nbsp block | repeatable
  - [`default`](#default) &nbsp any | optional
  - [`description`](#description) &nbsp string | optional
  - [`nullable`](#nullable) &nbsp boolean | optional
  - [`sensitive`](#sensitive) &nbsp boolean | optional
  - [`type`](#type) &nbsp type constraint | required

## Complete configuration example

> The following example demonstrates an input block with all configuration options specified, a second input block, and a policy block that references inputs:

Reviewer 2: ```suggestion
The following example demonstrates an `input` block with all configuration options specified, a second `input` block, and a `policy` block that references inputs:
```

  description = "The environment name (For example: dev, staging, prod)"
  default     = "dev"
}

resource_policy "aws_instance" "instance_type_check" {
  enforce {
    condition     = core::contains(input.allowed_instance_types, attrs.instance_type)
    error_message = "Instance type ${attrs.instance_type} is not in the allowed list for the "${input.environment}" environment: ${core::join(", ", input.allowed_instance_types)}"
  }
}
```

</CodeBlockConfig>

## Specification

An `input` block supports the following configuration.

### Label

The input block label gives a local name to the input variable. Must be unique within the policy file. Reference input variables with the syntax `input.<label>`.

### `default`

The `default` attribute declares provides a default value for the input variable. If omitted and no value is provided by the policy configuration, the value is null, and Terraform policy errors unless `nullable` is set to `true`.

- Data type: any
- Default: None
- Example: [String input with default](#string-input-with-default) 

### `description`

A description of the input variable's purpose. 

- Data type: String
- Default: None

### `nullable`

> Whether the input accepts a null value. If set to `false`, either the input block must provide a default value, or the value must be set by the policy configuration, or Terraform policy will error.

Reviewer 2: ```suggestion
Whether the input accepts a null value. If set to `false`, either the `input` block must provide a default value or the value must be set by the policy configuration, otherwise Terraform policy returns an error.
```

- Data type: Boolean
- Default: `false`

### `sensitive`

Whether the input variable contains sensitive data.

- Data type: Boolean
- Default `false`

### `type`

The type constraint for the input variable. Input variables support the [same type constraints](/terraform/language/expressions/types#types) as Terraform.

- Data type: Type constraint
- Required

- Data type: Type constraint

> - Required

Reviewer 2: ```suggestion
```
Dup

> - Required

Reviewer 3: Duplicated?
